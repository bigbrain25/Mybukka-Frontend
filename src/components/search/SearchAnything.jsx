import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import MarkIcon from 'Icons/Remark';
import { connect } from 'react-redux';
import Navlink from 'Components/navlink/Navlink';
import searchAnything from 'Redux/searchAnything';
import InputField from 'Components/input/InputField';
import Magnifier from '../icons/Magnifier';
import { ReusableButton, ReusableDropdown, ReusableWrapper }
  from '../common-navs/ReusableNavElements';
import data from './data.json';
import './searchanything.scss';

const { searchData } = data;

const SearchInputField = ({ handleChange, value }) => (
  <InputField
    type="text"
    name="searchLocation"
    placeholderText="Search for anything..."
    classNames="searchanything-text-field inherit-bg searchlocation"
    handleChange={handleChange}
    autoComplete="off"
    value={value}
  />
);

const DefaultSearchCategories = () => (
  <div className="search-padding">
    <div className="search-result-head">
      cuisines
    </div>
    <div className="row">
      {data.defaultSearch.map(searchList => (
        <Navlink
          key={`default-search-${searchList.split(' ').join('-')}}`}
          href={`/search?q=${searchList.split(' ').join('-')}`}
          classNames="col-6 defualt-search-category pointer text-dark"
        >
          <div
            className="col-12 px-0 defualt-search-category"
          >
            {searchList}
          </div>
        </Navlink>
      ))}
    </div>
  </div>
);

const SearchHead = ({ title }) => (
  <div className="search-result-head p-3">
    {title}
  </div>
);

const SearchItem = ({ searchObj }) => (
  <Navlink
    key={`search-item-${searchObj.title.split(' ').join('-')}-10`}
    href={`/search?q=${searchObj.title.split(' ').join('-')}`}
    classNames="search-link pointer"
  >
    <div className="row px-3 py-1 pointer">
      <div className="col-12 text-dark search-category">
        {searchObj.title}
        <span className="icon">
          <MarkIcon />
        </span>
      </div>
      <div className="col-12 text-muted font-size14">{searchObj.location}</div>
    </div>
  </Navlink>
);

const ViewAllSearch = ({ searchText }) => (
  <Fragment>
    <hr />
    <div className="row px-3 pt-1 pb-2 pointer">
      <div className="col-12 text-color search-text">
        View all results for {'"'}{searchText}{'"'}
      </div>
    </div>
  </Fragment>
);

const SearchTextMatch = ({ matchText }) => (
  matchText &&
  <div className="row px-3 pt-1 pb-2 pointer">
    <div className="col-10 text-muted search-text">
      {matchText}
    </div>
    <span className="pb-3 icon">
      <Magnifier />
    </span>
  </div>
);

const headerTexts = ['for delivery', 'for pickup', 'items'];

const lists = ['delivery', 'pickup', 'items'];

const SearchLists = ({ item }) => (
  <Fragment>
    {data.searchData.map(search => (
      search.items.map((itm) => {
        if (itm.includes(item)) {
          return (
            <Navlink
              classNames="link"
              key={`search-lists-${itm}-${item.split(' ').join('-')}`}
              href={`/search?q=${item.split(' ').join('-')}`}
            >
              <SearchTextMatch matchText={itm} />
            </Navlink>
          );
        }
        return null;
      })
    ))}

    {lists.map((list, index) => (
      <Fragment key={`fragment-list-${list}`}>
        <hr className="my-1" />
        <SearchHead title={headerTexts[index]} />
        {searchData.map((search) => {
          if (search.type === list) {
            return <SearchItem key={`search-type-${list}-${JSON.stringify(search)}`} searchObj={search} />;
          }
          return null;
        })}
      </Fragment>
    ))}
    <ViewAllSearch searchText={item} />
  </Fragment>
);

const SearchAnything = (props) => {
  const { handleSearch } = props;
  const [search, setSearch] = useState('');

  const handleChange = ({ target: { value } }) => {
    handleSearch(value);
    setSearch(value);
  };

  return (
    <ReusableWrapper>
      <ReusableButton classNames="inherit-bg" {...props}>
        <span className="current-location-button-icon custom-mt-minus19">
          <Magnifier />
        </span>
        <div>
          <div className="current-location-button-text">
            <SearchInputField value={search} handleChange={handleChange} />
          </div>
        </div>
      </ReusableButton>
      {props.focus &&
      <ReusableDropdown classNames="search-anything-scene">
        <div className="search-result-content">
          <div>
            {search === '' ?
              <DefaultSearchCategories />
              : <SearchLists item={search} />
            }
          </div>
        </div>
      </ReusableDropdown>
      }
    </ReusableWrapper>
  );
};

export default connect(null,
  { handleSearch: searchAnything }
)(SearchAnything);

SearchLists.defaultProps = {
  searchText: ''
};

ViewAllSearch.propTypes = {
  searchText: PropTypes.string.isRequired
};

SearchAnything.defaultProps = {
  handleSearch: () => {},
};

SearchAnything.propTypes = {
  handleSearch: PropTypes.func,
  focus: PropTypes.bool.isRequired
};

SearchInputField.propTypes = {
  handleChange: PropTypes.func.isRequired
};

SearchHead.propTypes = {
  title: PropTypes.string.isRequired
};

SearchLists.defaultProps = {
  item: ''
};

SearchLists.propTypes = {
  item: PropTypes.string,
};

SearchItem.propTypes = {
  searchObj: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.arrayOf(
        PropTypes.string
      )
    ])
  ).isRequired
};
