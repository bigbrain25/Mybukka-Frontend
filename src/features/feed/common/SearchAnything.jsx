import React, { Fragment } from 'react';

import Magnifier from 'Components/icons/Magnifier';
import InputField from 'Components/input/InputField';

import './searchAnything.scss';

const SearchLocation = () => (
  <Fragment>
    <div
      className="input-group address-input-section
      feed-address-input-section d-inline-flex ml-3"
    >
      <div className="input-group-prepends">
        <span className="nav-magnifier">
          <Magnifier />
        </span>
      </div>
      <InputField
        type="text"
        name="search"
        placeholderText="Search for anything..."
        classNames="text-field form-control"
        handleFocus={() => {}}
        handleChange={() => {}}
      />
    </div>
  </Fragment>
);

export default SearchLocation;
