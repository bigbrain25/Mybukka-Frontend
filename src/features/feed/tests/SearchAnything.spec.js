import React from 'react';

import SearchAnything from '../common/SearchAnything';

describe('SearchAnything FeedPage component', () => {
  const wrapper = mount(<SearchAnything />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
