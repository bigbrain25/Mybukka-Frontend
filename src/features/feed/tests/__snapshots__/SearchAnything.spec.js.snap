// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAnything FeedPage component renders properly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <SearchLocation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="input-group address-input-section feed-address-input-section d-inline-flex ml-3"
      >
        <div
          class="input-group-prepends"
        >
          <span
            class="nav-magnifier"
          >
            <svg
              class="magnifier"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <defs>
                <path
                  d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                  id="icon-search_svg__a"
                />
              </defs>
              <use
                fill="currentColor"
                fill-rule="evenodd"
                opacity=".9"
                xlink:href="#icon-search_svg__a"
              />
            </svg>
          </span>
        </div>
        <input
          class="text-field form-control"
          name="search"
          placeholder="Search for anything..."
          type="text"
          value=""
        />
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-group-prepends"
          >
            <span
              className="nav-magnifier"
            >
              <search />
            </span>
          </div>,
          <InputField
            classNames="text-field form-control"
            defaultValue=""
            handleChange={[Function]}
            handleFocus={[Function]}
            name="search"
            placeholderText="Search for anything..."
            type="text"
          />,
        ],
        "className": "input-group address-input-section feed-address-input-section d-inline-flex ml-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="input-group-prepends"
          >
            <span
              class="nav-magnifier"
            >
              <svg
                class="magnifier"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <defs>
                  <path
                    d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                    id="icon-search_svg__a"
                  />
                </defs>
                <use
                  fill="currentColor"
                  fill-rule="evenodd"
                  opacity=".9"
                  xlink:href="#icon-search_svg__a"
                />
              </svg>
            </span>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="nav-magnifier"
            >
              <search />
            </span>,
            "className": "input-group-prepends",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <span
                class="nav-magnifier"
              >
                <svg
                  class="magnifier"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <defs>
                    <path
                      d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                      id="icon-search_svg__a"
                    />
                  </defs>
                  <use
                    fill="currentColor"
                    fill-rule="evenodd"
                    opacity=".9"
                    xlink:href="#icon-search_svg__a"
                  />
                </svg>
              </span>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <search />,
                "className": "nav-magnifier",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": Object {
                    "instance": <svg
                      class="magnifier"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <defs>
                        <path
                          d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                          id="icon-search_svg__a"
                        />
                      </defs>
                      <use
                        fill="currentColor"
                        fill-rule="evenodd"
                        opacity=".9"
                        xlink:href="#icon-search_svg__a"
                      />
                    </svg>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <defs>
                          <path
                            d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                            id="icon-search_svg__a"
                          />
                        </defs>,
                        <use
                          fill="currentColor"
                          fillRule="evenodd"
                          opacity=".9"
                          xlinkHref="#icon-search_svg__a"
                        />,
                      ],
                      "className": "magnifier",
                      "height": "16",
                      "viewBox": "0 0 16 16",
                      "width": "16",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": <defs>
                          <path
                            d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                            id="icon-search_svg__a"
                          />
                        </defs>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <path
                            d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                            id="icon-search_svg__a"
                          />,
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": <path
                              d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                              id="icon-search_svg__a"
                            />,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "d": "M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z",
                              "id": "icon-search_svg__a",
                            },
                            "ref": null,
                            "rendered": Array [
                              undefined,
                            ],
                            "type": "path",
                          },
                        ],
                        "type": "defs",
                      },
                      Object {
                        "instance": <use
                          fill="currentColor"
                          fill-rule="evenodd"
                          opacity=".9"
                          xlink:href="#icon-search_svg__a"
                        />,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "fill": "currentColor",
                          "fillRule": "evenodd",
                          "opacity": ".9",
                          "xlinkHref": "#icon-search_svg__a",
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "use",
                      },
                    ],
                    "type": "svg",
                  },
                  "type": [Function],
                },
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "classNames": "text-field form-control",
            "defaultValue": "",
            "handleChange": [Function],
            "handleFocus": [Function],
            "name": "search",
            "placeholderText": "Search for anything...",
            "type": "text",
          },
          "ref": null,
          "rendered": Object {
            "instance": <input
              class="text-field form-control"
              name="search"
              placeholder="Search for anything..."
              type="text"
              value=""
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "text-field form-control",
              "defaultValue": "",
              "name": "search",
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "Search for anything...",
              "type": "text",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "input",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="input-group address-input-section feed-address-input-section d-inline-flex ml-3"
        >
          <div
            class="input-group-prepends"
          >
            <span
              class="nav-magnifier"
            >
              <svg
                class="magnifier"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <defs>
                  <path
                    d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                    id="icon-search_svg__a"
                  />
                </defs>
                <use
                  fill="currentColor"
                  fill-rule="evenodd"
                  opacity=".9"
                  xlink:href="#icon-search_svg__a"
                />
              </svg>
            </span>
          </div>
          <input
            class="text-field form-control"
            name="search"
            placeholder="Search for anything..."
            type="text"
            value=""
          />
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="input-group-prepends"
            >
              <span
                className="nav-magnifier"
              >
                <search />
              </span>
            </div>,
            <InputField
              classNames="text-field form-control"
              defaultValue=""
              handleChange={[Function]}
              handleFocus={[Function]}
              name="search"
              placeholderText="Search for anything..."
              type="text"
            />,
          ],
          "className": "input-group address-input-section feed-address-input-section d-inline-flex ml-3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="input-group-prepends"
            >
              <span
                class="nav-magnifier"
              >
                <svg
                  class="magnifier"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <defs>
                    <path
                      d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                      id="icon-search_svg__a"
                    />
                  </defs>
                  <use
                    fill="currentColor"
                    fill-rule="evenodd"
                    opacity=".9"
                    xlink:href="#icon-search_svg__a"
                  />
                </svg>
              </span>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="nav-magnifier"
              >
                <search />
              </span>,
              "className": "input-group-prepends",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <span
                  class="nav-magnifier"
                >
                  <svg
                    class="magnifier"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <defs>
                      <path
                        d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                        id="icon-search_svg__a"
                      />
                    </defs>
                    <use
                      fill="currentColor"
                      fill-rule="evenodd"
                      opacity=".9"
                      xlink:href="#icon-search_svg__a"
                    />
                  </svg>
                </span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <search />,
                  "className": "nav-magnifier",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": Object {
                      "instance": <svg
                        class="magnifier"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <defs>
                          <path
                            d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                            id="icon-search_svg__a"
                          />
                        </defs>
                        <use
                          fill="currentColor"
                          fill-rule="evenodd"
                          opacity=".9"
                          xlink:href="#icon-search_svg__a"
                        />
                      </svg>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <defs>
                            <path
                              d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                              id="icon-search_svg__a"
                            />
                          </defs>,
                          <use
                            fill="currentColor"
                            fillRule="evenodd"
                            opacity=".9"
                            xlinkHref="#icon-search_svg__a"
                          />,
                        ],
                        "className": "magnifier",
                        "height": "16",
                        "viewBox": "0 0 16 16",
                        "width": "16",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": <defs>
                            <path
                              d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                              id="icon-search_svg__a"
                            />
                          </defs>,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": <path
                              d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                              id="icon-search_svg__a"
                            />,
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": <path
                                d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                                id="icon-search_svg__a"
                              />,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "d": "M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z",
                                "id": "icon-search_svg__a",
                              },
                              "ref": null,
                              "rendered": Array [
                                undefined,
                              ],
                              "type": "path",
                            },
                          ],
                          "type": "defs",
                        },
                        Object {
                          "instance": <use
                            fill="currentColor"
                            fill-rule="evenodd"
                            opacity=".9"
                            xlink:href="#icon-search_svg__a"
                          />,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "fill": "currentColor",
                            "fillRule": "evenodd",
                            "opacity": ".9",
                            "xlinkHref": "#icon-search_svg__a",
                          },
                          "ref": null,
                          "rendered": Array [
                            undefined,
                          ],
                          "type": "use",
                        },
                      ],
                      "type": "svg",
                    },
                    "type": [Function],
                  },
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "classNames": "text-field form-control",
              "defaultValue": "",
              "handleChange": [Function],
              "handleFocus": [Function],
              "name": "search",
              "placeholderText": "Search for anything...",
              "type": "text",
            },
            "ref": null,
            "rendered": Object {
              "instance": <input
                class="text-field form-control"
                name="search"
                placeholder="Search for anything..."
                type="text"
                value=""
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "text-field form-control",
                "defaultValue": "",
                "name": "search",
                "onChange": [Function],
                "onFocus": [Function],
                "placeholder": "Search for anything...",
                "type": "text",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "input",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
